feat(HACBS-1552): add upload_sbom script

mmalina opened the pull request:
This script will push sbom data to Pyxis.

The script is ready for review. Unit tests will be added once first iteration of review is done to avoid wasting time in case significant refactoring comes out of the review.

For input, you'll need a Pyxis image id and an sbom. For how to get an sbom for an image url, see here: https://github.com/hacbs-release/release-bundles/pull/92
You'll need cosign:
- How to install: https://docs.sigstore.dev/cosign/installation/
- And then run something like: `cosign download sbom --output-file my-sbom.json $IMAGE_URL`

Then run the script, e.g.:
```
$ ./pyxis/upload_sbom.py --pyxis-graphql-api https://graphql.pyxis.qa.engineering.redhat.com/graphql/ --image-id 64020cafb94be12a048de276 --sbom-path ~/tmp/sbom/sbom-cyclonedx.json
```

johnbieren reviewed (commented):
Can you add return type and doc strings to all the functions? Some have them, some don't

mmalina commented on pyxis/upload_sbom.py:
```diff
@@ -0,0 +1,361 @@
+This consists of two steps:
+1. Create a ContentManifest in Pyxis, referencing the image ID provided
+   If a manifest already exists, creation is skipped.
+2. For each sbom compotent, create a ContentManifestComponent in Pyxis,
```
fixed, thanks

theflockers reviewed (dismissed):
LGTM. Tests not required but welcome.

mmalina commented:
> LGTM. Tests not required but welcome.

Thanks. I added unit tests. On top of that, I split out a new function - see the second commit.

@theflockers @davidmogar @johnbieren @happybhati please review the new changes - second and third commit - thanks.

johnbieren commented on pyxis/test_upload_sbom.py:
```diff
@@ -0,0 +1,416 @@
+    assert count == 2
+
+
+def test_get_existing_component_count__no_components():
```
I think a test with `image = {"foo": []}` or something can be nice. So it is not empty, just doesn't contain content_manifest_components. Obviously this will pass given the code, but helps prevent future breakage

johnbieren reviewed (commented):
Overall really good tests, just a few nit picks

mmalina commented on pyxis/test_upload_sbom.py:
```diff
@@ -0,0 +1,416 @@
+    mock_get_existing_bom_refs,
+    mock_create_content_manifest_component,
+):
+    """Creation of manifest and the first component is skipped"""
```
Yep, that's better, thanks. Fixed.
